### PR TITLE
[SPARK-28050][SQL]DataFrameWriter support insertInto a specific table partition

### PR DESCRIPTION
**Pull request by Le-Dai**
### What changes were proposed in this pull request?
support insertInto a specific table partition

### Why are the changes needed?
make the api more kind

### Does this PR introduce any user-facing change?
no

### How was this patch tested?
use api write data to partioned hive table 
df.write.insertInto(ptTableName, "pt1='2018',pt2='0601'")

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 1]
```diff
@@ -338,6 +338,23 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
...
+  /**
+    * Inserts the content of the `DataFrame` to the specific table partition.
+    *
+    * {{{
+    *    scala> Seq((3, 4)).toDF("j", "i").write.insertInto(ptTableName,"pt1='0101',pt2='0202'")
+    * }}}
+    *
+    * @since 3.0
+    */
+  def insertInto(tableName: String,partionInfo: String): Unit = {
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: nit: add space `String, partionInfo`

**Comment by maropu** [comment]
Can you add some tests before accepting Jenkins?

**Comment by MrDLontheway** [comment]
@maropu  How is the general format of the spark code?  the last commit like ok

**Comment by maropu** [comment]
you failed to rebase...

**Comment by MrDLontheway** [comment]
sorry,I am a community novice , should i create a new PR ?

**Comment by maropu** [comment]
you can fix it by a git command, but reopening is easier I think.

**Pull request closed without merging**